diff --git a/node_modules/@react-native-community/geolocation/android/build.gradle b/node_modules/@react-native-community/geolocation/android/build.gradle
index aab4a31..8f552ab 100644
--- a/node_modules/@react-native-community/geolocation/android/build.gradle
+++ b/node_modules/@react-native-community/geolocation/android/build.gradle
@@ -40,6 +40,11 @@ android {
     targetSdkVersion getExtOrIntegerDefault('targetSdkVersion')
     buildConfigField "boolean", "IS_NEW_ARCHITECTURE_ENABLED", isNewArchitectureEnabled().toString()
   }
+
+  buildFeatures {
+    buildConfig true
+  }
+  
   buildTypes {
     release {
       minifyEnabled false
