diff --git a/node_modules/@react-native-seoul/naver-login/android/build.gradle b/node_modules/@react-native-seoul/naver-login/android/build.gradle
index f68c3cc..2fa47ff 100644
--- a/node_modules/@react-native-seoul/naver-login/android/build.gradle
+++ b/node_modules/@react-native-seoul/naver-login/android/build.gradle
@@ -18,6 +18,7 @@ apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
 
 android {
+  namespace 'com.dooboolab.naverlogin'
   compileSdkVersion getExtOrDefault('compileSdkVersion').toInteger()
   buildToolsVersion getExtOrDefault('buildToolsVersion')
 
