diff --git a/node_modules/react-native-date-picker/android/build.gradle b/node_modules/react-native-date-picker/android/build.gradle
index fda6003..f21df0e 100644
--- a/node_modules/react-native-date-picker/android/build.gradle
+++ b/node_modules/react-native-date-picker/android/build.gradle
@@ -13,6 +13,7 @@ def safeExtGet(prop, fallback) {
 }
 
 android {
+    namespace "com.henninghall.date_picker"
     compileSdkVersion safeExtGet('compileSdkVersion', 29)
     buildToolsVersion safeExtGet('buildToolsVersion', "25.0.3")
 
@@ -32,6 +33,11 @@ android {
             }
         }
     }
+
+    buildFeatures {
+    buildConfig true
+    }
+
     buildTypes {
         release {
             minifyEnabled false
